src/stream.c

```diff
@@ -146,19 +146,39 @@ avifBool avifROStreamReadU64(avifROStream * stream, uint64_t * v)
     return AVIF_TRUE;
 }
 
-// Override of avifROStreamReadBits() for convenient uint8_t output.
-avifBool avifROStreamReadBits8(avifROStream * stream, uint8_t * v, size_t bitCount)
+avifBool avifROStreamSkipBits(avifROStream * stream, size_t bitCount)
 {
-    assert(bitCount <= sizeof(*v) * 8);
-    uint32_t v32;
-    if (!avifROStreamReadBits(stream, &v32, bitCount)) {
-        return AVIF_FALSE;
+    if (stream->numUsedBitsInPartialByte != 0) {
+        assert(stream->numUsedBitsInPartialByte < 8);
+        const size_t padding = AVIF_MIN(8 - stream->numUsedBitsInPartialByte, bitCount);
+        stream->numUsedBitsInPartialByte = (stream->numUsedBitsInPartialByte + padding) % 8;
+        bitCount -= padding;
     }
-    *v = (uint8_t)v32;
+    const size_t num_bytes = (bitCount + 7) / 8;
+    AVIF_CHECK(avifROStreamSkip(stream, num_bytes));
+    stream->numUsedBitsInPartialByte = bitCount % 8;
+    return AVIF_TRUE;
+}
+
+avifBool avifROStreamReadBitsU8(avifROStream * stream, uint8_t * v, size_t bitCount)
+{
+    assert(bitCount <= sizeof(*v) * 8);
```

Comment:
Optional: It seems that these assertions should be replaced by error checking (returning `AVIF_FALSE` on failure).

Reply:
Done in https://github.com/AOMediaCodec/libavif/pull/2457.